packages/wallet-ui/src/components/ui/organism/SideBar/SideBar.view.tsx

```diff
@@ -113,13 +114,14 @@ export const SideBarView = ({ address }: Props) => {
         <AccountImageStyled address={address} connected={wallet.connected} />
       </AccountDetails>
 
-      <AccountLabel>My account</AccountLabel>
+      <AccountLabel>Account {addressIndex + 1} </AccountLabel>
       <RowDiv>
         <InfoIcon onClick={() => setInfoModalOpen(true)}>i</InfoIcon>
         <AccountSwitchModal
           currentAddress={address}
           starkName={starkName}
-          accounts={accounts}
+          accounts={accounts.map((account) => account.address)}
+          accountsIndex={accounts.map((account) => account.addressIndex)}
```

Comment:
if you already update the slice from storing the account address to the whole account object
then why you need to do 2 pass to loop the accounts to get the address and index, seem not make sense

```suggestion
          accounts={accounts)}
```